src/java.desktop/windows/classes/sun/awt/shell/Win32ShellFolder2.java

```diff
@@ -1200,12 +1200,15 @@ public Image getIcon(int width, int height) {
      * Gets an icon from the Windows system icon list as an {@code Image}
      */
     static Image getSystemIcon(SystemIcon iconType) {
+        Image icon = null;
         long hIcon = getSystemIcon(iconType.getIconID());
-        Image icon = makeIcon(hIcon);
-        if (icon != null && LARGE_ICON_SIZE != icon.getWidth(null)) {
-            icon = new MultiResolutionIconImage(LARGE_ICON_SIZE, icon);
+        if (hIcon != 0) {
```

Comment:
I suggest inverting the condition:

```java
if (hIcon == 0) {
    return null;
}

Image icon = makeIcon(hIcon);
```

The code has less indentation and has better readability, in my opinion.

Reply:
Ah, I see the current style that you use follows the code style in `getShell32Icon`. 

Either go with the current style, or change the condition in `getShell32Icon` so that both methods use the same code flow.